### DISCUSSION TITLE
Is Cefsharp version for MAC OSX available?

### DISCUSSION THREAD

**Discussion by assopri** [cefsharp/CefSharp#3960] [Q&A]
Hello! In our Windows application (.NET 4.5.2) we use Cefsharp 91.1.230 for headless and headfull browser automation. We load webpages, call javascript on the webpage (to imitate clicks etc), emulate mouse and keyboard user input. Now we want to port this application to MAC (I mean just to create version for MAC). Is any Cefsharp version available for this purpose?

**Comment by amaitland** [comment]
CefSharp is built with Microsoft Visual C++ which is Windows only. So CefSharp only runs on Windows.

**Reply by assopri** [reply]
Thank you for reply! What about crossplatform CefGlue or CefGlue.Avalonia? Does it provide less capabilities, than CefSharp has?

**Reply by amaitland** [reply]
> What about crossplatform CefGlue or CefGlue.Avalonia?

As WinForms and WPF are both windows only you'll need to port your application to a cross platform UI framework. Avalonia is the framework I'd look at first.

CefGlue is also CEF based, so you can fundamentally do the same things, you'll likely have to implement some of the functionality yourself. You can of course reference the CefSharp source code to get some pointers.

There is also https://github.com/CefNet/CefNet which has a Avalonia implementation.
